query/benchmarker.go

```diff
@@ -141,11 +148,20 @@ func (b *BenchmarkRunner) Run(queryPool *sync.Pool, processorCreateFn ProcessorC
 	// Launch the stats processor:
 	go b.sp.process(b.Workers)
 
+	var requestRate = rate.Limit(math.MaxFloat64)
```

Comment:
to make this more testable, we can add an unexported method for getting the rateLimiter based on LimitRPS and Workers:
```go
func getRateLimiter(limitRPS uint64, workers uint) *rate.Limiter
```

Then it can be tested independently to make sure it returns the right things when limit is 0 or not.